File: Behavioral/ChainOfResponsibility/Python/basic_support.py
# basic_support.py

from support_handler import SupportHandler

class BasicSupport(SupportHandler):
  def handle_request(self, request):
    if request == "Basic Issue":
      print(f"BasicSupport: Handling request - {request}")
    elif self._next_handler:
      self._next_handler.handle_request(request)
    else:
      print(f"No handler available for - {request}")

File: Behavioral/ChainOfResponsibility/CSharp/BasicSupport.cs
// BasicSupport.cs

using System;

public class BasicSupport : SupportHandler {
  public override void HandleRequest(string request) {
    if (request == "Basic Issue") {
      Console.WriteLine($"BasicSupport: Handling request - {request}");
    } else if (NextHandler != null) {
      NextHandler.HandleRequest(request); // Only pass if not handled
    } else {
      Console.WriteLine($"No handler available for - {request}");
    }
  }
}
